runtime/runtime/src/lib.rs

```diff
@@ -342,16 +344,32 @@ impl Runtime {
         // TODO(#8806): Support compute costs for actions. For now they match burnt gas.
         result.compute_usage = exec_fees;
         let account_id = &receipt.receiver_id;
-        let is_the_only_action = actions.len() == 1;
         let is_refund = receipt.predecessor_id.is_system();
+
+        #[cfg(not(feature = "protocol_feature_nonrefundable_transfer_nep491"))]
+        let is_the_only_action = actions.len() == 1;
+
+        #[cfg(feature = "protocol_feature_nonrefundable_transfer_nep491")]
+        let only_transfers = actions.iter().all(|action| {
+            matches!(action, Action::Transfer(_) | Action::NonrefundableStorageTransfer(_))
+        });
+
+        #[cfg(not(feature = "protocol_feature_nonrefundable_transfer_nep491"))]
+        let implicit_account_creation_eligible = is_the_only_action && !is_refund;
+
+        #[cfg(feature = "protocol_feature_nonrefundable_transfer_nep491")]
+        let implicit_account_creation_eligible = only_transfers && !is_refund;
```

Comment:
Doesn't this new code accept transactions with multiple transfers to create implicit accounts? I think today a double transfer would fail if the account doesn't exist, yet.

If that is indeed what happens, you should use `checked_feature!` here instead of manual cfg.
Because during the protocol upgrade, some nodes will run the new code and some will use the old code, and we can't have them disagree on what is a valid account creation and what isn't.

But really, I think we should keep the behaviour the same. A refundable transfer followed by another transfer should keep failing unless there is a good reason to make that protocol change.

Reply:
For simplicity, I would just stick with the previous behavior, which is to require exactly one transfer action on implicit account creation. This was agreed in NEP and we can easily allow for multiple transfers later.
I would document that `NonrefundableStorageTransfer` has to be the only action in a receipt for implicit account creation.
cc @birchmd

Reply:
Sure, as long as it's documented I think either behaviour is fine.

Reply:
Makes sense to me, too.